### PR TITLE
Change 'type' in `CREATE DATA LINK` from full class path to shortened name [HZ-2179]

### PR DESCRIPTION
**Pull request by Fly-Style**
At the same time, I also intended to preserve option for test class

### CODE REVIEW

**Comment by Fly-Style** [comment]
run-sql-only

**Comment by Fly-Style** [comment]
run-sql-only

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/infoschema/DataLinksUtil.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ */
+
+package com.hazelcast.jet.sql.impl.connector.infoschema;
+
+import com.hazelcast.datalink.JdbcDataLink;
+
+import java.util.Locale;
+
+public final class DataLinksUtil {
+    public static final String DUMMY_CONNECTOR_TYPE = "com.hazelcast.datalink.impl.DataLinkTestUtil$DummyDataLink";
```
[the remark this answers is not in the document]
Comment (Fly-Style) [reply hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/infoschema/DataLinksUtil.java:24]: That is a special case to allow testing data links with 'dummy' data link which does nothing and doesn't have any resources.

[the remark this answers is not in the document]
Comment (k-jamroz) [reply hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/infoschema/DataLinksUtil.java:24]: I do not mean only DUMMY_CONNECTOR_TYPE (which BTW could by handled nicely by ServiceLoader and be available only in tests). We hardcode all other connectors as well.

[the remark this answers is not in the document]
Comment (Fly-Style) [reply hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/infoschema/DataLinksUtil.java:24]: Sorry, but I really don't understand why should we use `ServiceLoader` here. I need connector's class names to pass them as 'type' argument to `replaceSqlDataLink` method

[the remark this answers is not in the document]
Comment (k-jamroz) [reply hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/infoschema/DataLinksUtil.java:24]: you hardcoded all supported connectors in `DataLinksUtil`. It is not modular nor extensible. Connectors is the part that should be easily extensible, even if we are the only ones that will ever implement a connector. And SQL already uses `ServiceLoader` for the same purpose.

[the remark this answers is not in the document]
Comment (Fly-Style) [reply hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/infoschema/DataLinksUtil.java:24]: Ah, see it

[the remark this answers is not in the document]
Comment (k-jamroz) [reply hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/infoschema/DataLinksUtil.java]: there already existed `DataLinkRegistration` so I used that one. Probably somewhere we have or will need to have a mapping between data link classes and SqlConnectors but this is not the part of this PR.

**Comment by k-jamroz** [comment]
run-lab-run

**Comment by Fly-Style** [comment]
run-lab-run

**Pull request merged**